agent/lib/kontena/helpers/weave_helper.rb

```diff
@@ -5,11 +5,22 @@ module Kontena
   module Helpers
     module WeaveHelper
 
-      # @return [Boolean]
-      def weave_running?
-        weave = Docker::Container.get('weave') rescue nil
-        return false if weave.nil?
-        weave.info['State']['Running'] == true
+      def network_adapter
+        Celluloid::Actor[:network_adapter]
+      end
+
+      def wait_weave_running?(timeout = 10, &block)
```

Comment:
The timeout should be longer than the expected Docker restart interval for the kontena-etcd and kontena-ipam.

It's probably more than 10s?